src/main/Config.cpp

```diff
@@ -1411,7 +1384,9 @@ Config::generateQuorumSetHelper(
                 "High quality validator {} must have redundancy of at least 3",
                 it->mName));
         }
-        innerSet.threshold = computeDefaultThreshold(innerSet, true);
+        innerSet.threshold = computeDefaultThreshold(
```

Comment:
`CRITICAL` is better than `HIGH`, so we should require at least 3 validators at that quality level